include/solvers/mpm_base.tcc

```diff
@@ -96,10 +96,44 @@ mpm::MPMBase<Tdim>::MPMBase(const std::shared_ptr<IO>& io) : mpm::MPM(io) {
   try {
     if (post_process_.at("vtk_statevars").is_array() &&
         post_process_.at("vtk_statevars").size() > 0) {
-      for (unsigned i = 0; i < post_process_.at("vtk_statevars").size(); ++i) {
-        std::string attribute =
-            post_process_["vtk_statevars"][i].template get<std::string>();
-        vtk_statevars_.emplace_back(attribute);
+      // Iterate over state_vars
+      for (const auto& svars : post_process_["vtk_statevars"]) {
+        // Initial phase id and state vars
+        unsigned phase_id = 0;
+        std::vector<std::string> state_var;
+
+        // If state variables are arranged as string
```

Comment:
Why not just set input JSON to a unified format, where they have to specify phase_id and variables?